### PR TITLE
Add displayURL and expandURL into Entity, and use them in Autolink.autoLinkEntities()

### PR DESCRIPTION
**Pull request by keitaf**
This is to support the same functionality as in
https://github.com/twitter/twitter-text-js/pull/36

### CODE REVIEW

File: src/com/twitter/Autolink.java [hunk 1]
```diff
@@ -93,11 +93,72 @@ private String autoLinkEntities(String text, List<Entity> entities) {
       switch(entity.type) {
         case URL:
           CharSequence url = escapeHTML(entity.getValue());
+          CharSequence linkText = url;
```
Comment (jsha) [inline src/com/twitter/Autolink.java:96]: Need an HTML escape here.

File: src/com/twitter/Autolink.java [hunk 2]
```diff
@@ -93,11 +93,72 @@ private String autoLinkEntities(String text, List<Entity> entities) {
...
+            //   …
+            // </span>
+            //
+            // Exception: pic.twitter.com images, for which expandedUrl = "https://twitter.com/#!/username/status/1234/photo/1
+            // For those URLs, display_url is not a substring of expanded_url, so we don't do anything special to render the elided parts.
+            // For a pic.twitter.com URL, the only elided part will be the "https://", so this is fine.
+            String displayURLSansEllipses = entity.displayURL.replace("…", "");
+            int diplayURLIndexInExpandedURL = entity.expandedURL.indexOf(displayURLSansEllipses);
+            if (diplayURLIndexInExpandedURL != -1) {
+              String beforeDisplayURL = entity.expandedURL.substring(0, diplayURLIndexInExpandedURL);
```
Comment (jsha) [inline src/com/twitter/Autolink.java:140]: Need to HTML escape beforeDisplayURL, afterDisplayURL, and displayURLSansEllipses.

File: src/com/twitter/Autolink.java (hunk 1 above)
Comment (keitaf) [inline src/com/twitter/Autolink.java:96]: It's already escaped in line 95.

File: src/com/twitter/Autolink.java (hunk 2 above)
Comment (keitaf) [inline src/com/twitter/Autolink.java:140]: Done!

**Comment by jsha** [comment]
Looks good to me!

**Pull request merged**